sslcontext-kickstart/src/main/java/nl/altindag/ssl/trustmanager/LoggingX509ExtendedTrustManager.java

```diff
@@ -37,58 +37,73 @@
     public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
-        logCertificate(CounterParty.SERVER, chain, authType, null, null);
-        super.checkServerTrusted(chain, authType);
+        checkTrusted(() -> super.checkServerTrusted(chain, authType), CounterParty.SERVER, chain, authType, null, null);
     }
 
     @Override
     public void checkServerTrusted(X509Certificate[] chain, String authType, Socket socket) throws CertificateException {
-        logCertificate(CounterParty.SERVER, chain, authType, socket, null);
-        super.checkServerTrusted(chain, authType, socket);
+        checkTrusted(() -> super.checkServerTrusted(chain, authType, socket), CounterParty.SERVER, chain, authType, socket, null);
     }
 
     @Override
     public void checkServerTrusted(X509Certificate[] chain, String authType, SSLEngine sslEngine) throws CertificateException {
-        logCertificate(CounterParty.SERVER, chain, authType, null, sslEngine);
-        super.checkServerTrusted(chain, authType, sslEngine);
+        checkTrusted(() -> super.checkServerTrusted(chain, authType, sslEngine), CounterParty.SERVER, chain, authType, null, sslEngine);
     }
 
-    private static void logCertificate(CounterParty counterParty, X509Certificate[] chain, String authType, Socket socket, SSLEngine sslEngine) {
+    private static void checkTrusted(TrustManagerRunnable runnable,
+                                     CounterParty counterParty,
+                                     X509Certificate[] chain,
+                                     String authType,
+                                     Socket socket,
+                                     SSLEngine sslEngine) throws CertificateException {
+
         String certificateChain = Arrays.toString(chain);
 
         Optional<String> classNameLogMessage = getClassnameOfEitherOrOther(socket, sslEngine)
-                .map(className -> ", while also using the " + className + "");
+                .map(className -> ", while also using the " + className);
         Optional<String> hostAndPortLogMessage = getHostAndPortOfEitherOrOther(socket, sslEngine)
                 .map(hostAndPort -> "[" + hostAndPort + "]");
 
         String logMessage = String.format(LOG_MESSAGE_TEMPLATE, counterParty, hostAndPortLogMessage.orElse(""), authType, classNameLogMessage.orElse(""), counterParty, certificateChain);
         LOGGER.debug(logMessage);
+
+        try {
+            runnable.run();
+            String okMessage = String.format(VALIDATION_PASSED_LOG_MESSAGE_TEMPLATE, counterParty, hostAndPortLogMessage.orElse(""), authType, classNameLogMessage.orElse(""));
+            SUCCESS_LOGGER.debug(okMessage);
+        } catch (CertificateException e) {
+            String nokMessage = String.format(VALIDATION_FAILED_LOG_MESSAGE_TEMPLATE, counterParty, hostAndPortLogMessage.orElse(""), authType, classNameLogMessage.orElse(""));
+            EXCEPTION_LOGGER.debug(nokMessage, e);
```

Comment:
how can users customize the message? should that be possible?

Reply:
You mean the content or the visibility?

Reply:
both

Reply:
The message cannot be adjusted. The visibility can be adjusted. The library  has slf4j api, so you need to add a logger implementation such as logback or slf4j simple and configure your logs with it